lib/assignDisabledRanges.js

```diff
@@ -259,18 +263,10 @@ module.exports = function assignDisabledRanges(root, result) {
 
 			if (ruleIsDisabled(ALL_RULES) && disabledRanges[ruleToEnable] === undefined) {
 				// Get a starting point from the where all rules were disabled
-				if (!disabledRanges[ruleToEnable]) {
```

Comment:
[note] If `ruleIsDisabled(ALL_RULES) && disabledRanges[ruleToEnable] === undefined` is `true`, `!disabledRanges[ruleToEnable]` should always be `true`. So, the `else` block should never be executed.